src/store/root.reducer.ts

```diff
@@ -1,21 +1,12 @@
-const initiatState = {
-  data: {},
-  petsToCompare: [],
-  search: {},
-};
+import { createReducer } from 'typesafe-actions';
 
-export const rootReducer = (state: object = initiatState, action: any) => {
-  switch (action.type) {
-    case 'PUT_PETS':
-      return {
-        ...state,
-        petsToCompare: action.payload.pets,
-      };
-    case 'SEARCH':
-      return {
-        search: action.payload,
-      };
-  }
+type State = {} | [];
+type Action = { type: 'FILTER'; payload: {} } | { type: 'PUT_PETS'; payload: any };
 
-  return state;
-};
+export const rootReducer = createReducer<State, Action>(0)
```

Comment:
You should not have any logic inside the `rootReducer`. 

I would expect to have something like this

```
expect const rootReducer = combineReducers({
  filterReducer: filterReducer,
  petsReducer: petsReducer
})
```